python/tritonserver/_api/_response.py

```diff
@@ -29,293 +29,37 @@
-                asyncio.run_coroutine_threadsafe(
-                    self._user_queue.put(response), self._loop
-                )
-            if flags == TRITONSERVER_ResponseCompleteFlag.FINAL:
-                del self._request
-                self._request = None
-        except Exception as e:
-            message = f"Catastrophic failure in response callback: {e}"
-            LogMessage(LogLevel.ERROR, message)
-            # catastrophic failure
-            raise e from None
-
-
-class ResponseIterator:
-    """Response iterator
-
-    Response iterators are returned from model inference methods and
-    allow users to process inference responses in the order they were
-    received for a request.
-
-    """
-
-    def __init__(
-        self,
-        model: Model,
-        request: TRITONSERVER_InferenceRequest,
-        user_queue: Optional[queue.SimpleQueue] = None,
-        raise_on_error: bool = False,
-    ):
-        """Initialize ResponseIterator
-
-        ResponseIterator objects are obtained from Model inference
-        methods and not instantiated directly. See `Model` documentation.
-
-        Parameters
-        ----------
-        model : Model
-            model associated with inference request
-        request : TRITONSERVER_InferenceRequest
-            Underlying C binding TRITONSERVER_InferenceRequest
-            object. Private.
-        user_queue : Optional[asyncio.Queue]
-            Optional user queue for responses in addition to internal
-            iterator queue.
-        raise_on_error : bool
-            if True response errors will be raised as exceptions.
-
-        """
-
-        self._queue = queue.SimpleQueue()
-        self._user_queue = user_queue
-        self._complete = False
-        self._request = request
-        self._model = model
-        self._raise_on_error = raise_on_error
-
-    def __iter__(self) -> ResponseIterator:
-        """Return response iterator.
-
-        Returns
-        -------
-        ResponseIterator
-
-        Examples
-        --------
-
-        responses = server.model("test").infer(inputs={"fp16_input":numpy.array([[1]],dtype=numpy.float16)})
-        for response in responses:
-           print(nummpy.from_dlpack(response.outputs["fp16_output"]))
-
-
-        """
-
-        return self
-
-    def __next__(self):
-        """Returns the next response received for a request
-
-        Raises
-        ------
-        response.error
-            If raise_on_error is set to True, response errors are
-            raised as exceptions
-        StopIteration
-            Indicates all responses for a request have been received.
-            Final responses may or may not include outputs and must be
-            checked.
-
-        """
-
-        if self._complete:
-            raise StopIteration
-        response = self._queue.get()
-        self._complete = response.final
-        if response.error is not None and self._raise_on_error:
```

Comment:
this seems missing from the new code?

Reply:
yes, this is added [Add raise on error to response iterators](https://github.com/triton-inference-server/core/pull/426/commits/898ef02f33e74822a6b175068f4ae7cdf13c62d4)